### PR TITLE
Added Improved Rest API

### PR DESCRIPTION
**Pull request by jniebuhr**
As requested in #12, I added an improved json api. I have however not implemented paging support yet.

### CODE REVIEW

File: wpc/api.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    except Exception as e:
+        app.logger.exception(e)
+        abort(500)
+
+@app.route('/api/v1/streamers/<name>/live')
+@crossdomain(origin='*', max_age=15)
+def api_streamers_live(name):
+    try:
+        streamer = Streamer.query.filter_by(reddit_username=name).first_or_404()
+        if streamer is None:
```
Comment (eleweek) [inline wpc/api.py]: unnecessary, since first_or_404() calls abort(404) if streamer isn't found

File: wpc/api.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    except Exception as e:
+        app.logger.exception(e)
+        abort(500)
+
+@app.route('/api/v1/streamers/<name>/completed')
+@crossdomain(origin='*', max_age=15)
+def api_streamers_past(name):
+    try:
+        streamer = Streamer.query.filter_by(reddit_username=name).first_or_404()
+        if streamer is None:
```
Comment (eleweek) [inline wpc/api.py]: unnecessary, since first_or_404() calls abort(404) if streamer isn't found

File: wpc/api.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    except Exception as e:
+        app.logger.exception(e)
+        abort(500)
+
+@app.route('/api/v1/streamers/<name>/upcoming')
+@crossdomain(origin='*', max_age=15)
+def api_streamers_upcoming(name):
+    try:
+        streamer = Streamer.query.filter_by(reddit_username=name).first_or_404()
+        if streamer is None:
```
Comment (eleweek) [inline wpc/api.py]: unnecessary, since first_or_404() calls abort(404) if streamer isn't found

File: wpc/api.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    except Exception as e:
+        app.logger.exception(e)
+        abort(500)
+
+@app.route('/api/v1/streamers/<name>')
+@crossdomain(origin='*', max_age=15)
+def api_streamers_view(name):
+    try:
+        streamer = Streamer.query.filter_by(reddit_username=name).first_or_404()
+        if streamer is None:
```
Comment (eleweek) [inline wpc/api.py]: unnecessary, since first_or_404() calls abort(404) if streamer isn't found

File: wpc/api.py [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+                       info={'status': 200})
+    except Exception as e:
+        app.logger.exception(e)
+        abort(500)
+
+@app.route('/api/v1/streams/<stream_id>')
+@crossdomain(origin='*', max_age=15)
+def api_streams_view(stream_id):
+    try:
+        st = Stream.query.get(stream_id)
```
Comment (eleweek) [inline wpc/api.py]: it is better to use get_or_404() here

**Comment by eleweek** [comment]
Ok, looks good to me!

**Comment by paked** [comment]
Awesome!

**Pull request merged**